doc/source/image_description/elements.rst

```diff
@@ -406,6 +421,190 @@ target_blocksize="number"
 
      blockdev --report $DEVICE
 
+target_removable="true|false":
+  Indicate if the target disk for oem images is deployed
+  to a removable device e.g a USB stick or not. This only
+  affects the EFI setup if requested and in the end avoids
+  the creation of a custom boot menu entry in the firmware
+  of the target machine. By default the target disk is
+  expected to be non-removable
+
+spare_part="number":
+  Request a spare partition right before the root partition
+  of the requested size. The attribute takes a size value
+  and allows a unit in MB or GB, e.g 200M. If no unit is given
+  the value is considered to be mbytes. A spare partition
+  can only be configured for the disk image types oem and vmx
+
+spare_part_mountpoint="dir_path":
+  Specify mount point for spare partition in the system.
+  Can only be configured for the disk image types oem and vmx
+
+spare_part_fs="btrfs|ext2|ext3|ext4|xfs":
+  Specify filesystem for spare partition in the system.
+  Can only be configured for the disk image types oem and vmx
+
+spare_part_fs_attributes="attribute_list":
+  Specify filesystem attributes for the spare partition.
+  Attributes can be specified as comma separated list.
+  Currently the attributes `no-copy-on-write` and `synchronous-updates`
+  are available. Can only be configured for the disk image
+  types oem and vmx
+
+spare_part_is_last="true|false":
+  Specify if the spare partition should be the last one in
+  the partition table. Can only be configured for the vmx
+  disk image type. By default the root partition is the last
+  one and the spare partition lives behind it. With this
```

Comment:
nit: probably this is a misunderstanding caused by my poor English, to me it is odd having a partition behind last one... shouldn't it be before? To my understanding if something is last, there isn't anything else behind it, just before it :P

Reply:
Surely not your English but me writing a wrong sentence :) Thanks for pointing this out. Of course if by default root is last, spare lives before not behind. I will fix this

Reply:
hope it makes more sense now :)